angular/projects/spark-angular/src/lib/components/sprk-dropdown/sprk-dropdown.component.spec.ts

```diff
@@ -68,6 +126,27 @@ describe('SprkDropdownComponent', () => {
     expect(dropdownTriggerElement.getAttribute('data-id')).toBeNull();
   });
 
+  it('should add data-analytics when analyticsString has a value', () => {
+    const testString = 'element-id';
+    wrapperComponent.analyticsString = testString;
+    fixture.detectChanges();
+    expect(dropdownTriggerElement.getAttribute('data-analytics')).toEqual(
+      testString,
+    );
+  });
+
+  it('should not add data-analytics when analyticsString has no value', () => {
+    wrapperComponent.analyticsString = null;
+    fixture.detectChanges();
+    expect(dropdownTriggerElement.getAttribute('data-analytics')).toBeNull();
+  });
+
+  it('should render open with isOpen true', () => {
+    wrapperComponent.isOpen = true;
+    fixture.detectChanges();
+    expect(dropdownComponent.isOpen).toEqual(true);
```

Comment:
I think what we want to test here is that the dropdown is open in the DOM and not necessarily that the input is set to be true.